### PR TITLE
ci: upload source archive

### PR DESCRIPTION
**Pull request by HashidaTKS**
Added a `Source` job for CI that uploads a source archive.

### CODE REVIEW

**Comment by HashidaTKS** [comment]
@kou  @komainu8 

Would you review this when you have time?

File: .github/workflows/package.yml [hunk 1]
```diff
@@ -10,6 +10,45 @@ on:
...
+            qemu-user-static \
+            ruby
+          sudo gem install bundler
+          bundle install
+      - name: Clone dependencies
+        run: |
+          cd ..
+          git clone --depth 1 https://github.com/groonga/groonga.git
+          cd groonga
+          git submodule update --init --force --recursive --depth 1
```
Comment (kou) [inline .github/workflows/package.yml:33]: Could you use the latest released archive instead?

File: .github/workflows/package.yml [hunk 2]
```diff
@@ -10,6 +10,45 @@ on:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Install dependencies
+        run: |
+          sudo apt update
+          sudo apt -y -V install \
+            qemu-user-static \
```
Comment (kou) [inline .github/workflows/package.yml:24]: Do we need this?
We don't use Docker image for arm64.

File: .github/workflows/package.yml [hunk 3]
```diff
@@ -10,6 +10,45 @@ on:
...
+          git submodule update --init --force --recursive --depth 1
+      - name: Build archive
+        run: |
+          rake dist
+
+      # Artifact
+      - uses: actions/upload-artifact@v2
+        with:
+          name: source
+          path: pgroonga-*.tar.gz
```
Comment (kou) [inline .github/workflows/package.yml:42]: Could you use this in `linux` job?

**Comment by komainu8** [comment]
Could you add `needs: source` in `linux` job?
Because `linux` job use a source archive that build in `source` job.

File: .github/workflows/package.yml (hunk 2 above)
Comment (HashidaTKS) [reply .github/workflows/package.yml:24]: Oh, is `qemu-user-static` installed for Docker image for arm64 at this context?
Ok, I will remove this.

Comment (kou) [reply .github/workflows/package.yml:24]: Yes.

File: .github/workflows/package.yml (hunk 1 above)
Comment (HashidaTKS) [reply .github/workflows/package.yml]: In my second thought, the archive file doesn't depend on groonga.
So I have remove this step.

Related to this, the linux job use not the latest released archive but the latest Groonga repository.

https://github.com/pgroonga/pgroonga/blob/master/.github/workflows/package.yml#L88

I will fix it in another PR.

File: .github/workflows/package.yml (hunk 2 above)
Comment (HashidaTKS) [reply .github/workflows/package.yml]: Thanks, fixed.

File: .github/workflows/package.yml (hunk 3 above)
Comment (HashidaTKS) [reply .github/workflows/package.yml:35]: Changed to use this archive,

**Comment by HashidaTKS** [comment]
@kou @komainu8 

Thank you, I have addressed your comments.
Please re-review this PR.

**Pull request merged**